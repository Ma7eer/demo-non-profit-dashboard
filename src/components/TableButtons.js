import React from 'react'
import PropTypes from 'prop-types'
import { Button, ButtonToolbar } from 'react-bootstrap'

const TableButtons = ({
  row,
  handleEdit,
  handleDelete,
  showDelete = true,
  showEdit = true
}) => {
  return showDelete && showEdit ? (
    <ButtonToolbar style={{ display: 'flex', justifyContent: 'center' }}>
      <Button
        bsSize="small"
        bsStyle="warning"
        onClick={() => handleEdit(row.original, row.index)}
      >
        <i className="glyphicon glyphicon-pencil" />
      </Button>
      <Button bsSize="small" bsStyle="danger" onClick={() => handleDelete(row)}>
        <i className="glyphicon glyphicon-trash" />
      </Button>
    </ButtonToolbar>
  ) : showDelete ? (
    <ButtonToolbar style={{ display: 'flex', justifyContent: 'center' }}>
      <Button bsSize="small" bsStyle="danger" onClick={() => handleDelete(row)}>
        <i className="glyphicon glyphicon-trash" />
      </Button>
    </ButtonToolbar>
  ) : (
    <ButtonToolbar style={{ display: 'flex', justifyContent: 'center' }}>
      <Button
        bsSize="small"
        bsStyle="warning"
        onClick={() => handleEdit(row.original, row.index)}
      >
        <i className="glyphicon glyphicon-pencil" />
      </Button>
    </ButtonToolbar>
  )
}

TableButtons.propTypes = {
  row: PropTypes.object,
  handleEdit: PropTypes.func,
  handleDelete: PropTypes.func,
  showDelete: PropTypes.bool,
  showEdit: PropTypes.bool
}

export default TableButtons
