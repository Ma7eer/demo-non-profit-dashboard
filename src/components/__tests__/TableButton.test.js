import '@testing-library/react/cleanup-after-each'
import React from 'react'
import { render } from '@testing-library/react'
import TableButtons from '../TableButtons'

test('TableButtons renders if no props given', () => {
  const { container } = render(<TableButtons />)
  expect(container.firstChild).toMatchSnapshot()
})

test('TableButtons renders only edit button', () => {
  const { container } = render(<TableButtons showDelete={false} />)
  expect(container.firstChild).toMatchSnapshot()
})

test('TableButtons renders only delete button', () => {
  const { container } = render(<TableButtons showEdit={false} />)
  expect(container.firstChild).toMatchSnapshot()
})
