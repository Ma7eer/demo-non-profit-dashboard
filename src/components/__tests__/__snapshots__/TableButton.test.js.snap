// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableButtons renders if no props given 1`] = `
<div
  class="btn-toolbar"
  role="toolbar"
  style="display: flex; justify-content: center;"
>
  <button
    class="btn btn-sm btn-warning"
    type="button"
  >
    <i
      class="glyphicon glyphicon-pencil"
    />
  </button>
  <button
    class="btn btn-sm btn-danger"
    type="button"
  >
    <i
      class="glyphicon glyphicon-trash"
    />
  </button>
</div>
`;

exports[`TableButtons renders only delete button 1`] = `
<div
  class="btn-toolbar"
  role="toolbar"
  style="display: flex; justify-content: center;"
>
  <button
    class="btn btn-sm btn-danger"
    type="button"
  >
    <i
      class="glyphicon glyphicon-trash"
    />
  </button>
</div>
`;

exports[`TableButtons renders only edit button 1`] = `
<div
  class="btn-toolbar"
  role="toolbar"
  style="display: flex; justify-content: center;"
>
  <button
    class="btn btn-sm btn-warning"
    type="button"
  >
    <i
      class="glyphicon glyphicon-pencil"
    />
  </button>
</div>
`;
